chalice/app.py

```diff
@@ -266,15 +292,42 @@ def __init__(self, body, headers=None, status_code=200):
         self.headers = headers
         self.status_code = status_code
 
-    def to_dict(self):
+    def to_dict(self, binary_types=None):
         body = self.body
-        if not isinstance(body, str):
+        if not isinstance(body, (str, bytes)):
             body = json.dumps(body, default=handle_decimals)
-        return {
+        response = {
             'headers': self.headers,
             'statusCode': self.status_code,
-            'body': body,
+            'body': body
         }
+        if binary_types is not None:
+            self._b64encode_body_if_needed(response, binary_types)
+        return response
+
+    def _b64encode_body_if_needed(self, response_dict, binary_types):
+        response_headers = CaseInsensitiveMapping(response_dict['headers'])
+        content_type = response_headers.get('content-type', '')
+        body = response_dict['body']
+
+        # Encode the response as binary if required.
+        # To keep json types working as expected they are encoded to a utf-8
+        # byte sequence if the content type is application/json before being
+        # converted to a base64 string.
+        if _matches_content_type(content_type, binary_types):
+            if _matches_content_type(content_type, ['application/json']):
```

Comment:
Cool. Could you add this information to the comment? Not sure if you need the whole thing but mainly clarifying the "working as expected" part.